src/eredis.hrl

```diff
@@ -15,12 +15,11 @@
 
 -define(NL, "\r\n").
 
--define(SOCKET_OPTS, [binary, {active, once}, {packet, raw}, {reuseaddr, false},
-                      {send_timeout, ?SEND_TIMEOUT},
-                      {keepalive, true}]).
-
--define(TCP_SOCKET_OPTS, ?SOCKET_OPTS).
--define(SSL_SOCKET_OPTS, ?SOCKET_OPTS).
+-define(TCP_SOCKET_OPTS, [binary, {active, once}, {packet, raw}, {reuseaddr, false},
+                          {send_timeout, 5000},
+                          {keepalive, true}]).
+-define(SSL_SOCKET_OPTS, [binary, {active, once}, {packet, raw}, {reuseaddr, false},
+                          {send_timeout, 5000},
+                          {keepalive, true}]).
```

Comment:
^ Why differentiate between TCP and SSL? Wouldn't a single `SOCKET_OPTS` be enough?

Reply:
I'm getting the feeling this is one of those some-tool-complains-otherwise situations and we end up serving the tool rather than having the tool serve us under a vile compromise. 

Is this the case? Does hank not like macro re-definitions? Is there some flag that we could get away with, to not have to bow down to The Machine?

Reply:
Well, since this an internal header and removing macros won't break the API (I hadn't noticed that earlier), I propose we get rid of `TCP_SOCKET_OPTS` and `SSL_SOCKET_OPTS` and just keep one `COMMON_SOCKET_OPTS`.

That way we avoid the vile compromise (assuming my hypothesis about hank, as I outlined in a previous comment, is true.) What do you think?

Reply:
We can keep the common approach. All I did was in the interest of the lib., not Hank. I thought "why are we assuming the elements are common?" and then I refactored it so it was explicit what we were doing.

Reply:
I force-pushed a squash commit.

Reply:
Ahh, fair enough! There I went again, seeing evil everywhere 😄